### DISCUSSION TITLE
Does Pytorch-Lightning support the task of federated learning?

### DISCUSSION THREAD

**Discussion by marsggbo** [Lightning-AI/pytorch-lightning#7727] [General]
Does Pytorch-Lightning support the task of federated learning? If so, any example provided? Thanks!

**Comment by kaushikb11** [comment]
@tchaton Your take?

**Comment by tchaton** [comment]
Hey @marsggbo,

We worked on a first integration with PySyft. Here is the notebook: https://github.com/OpenMined/PySyft/blob/dev/packages/syft/examples/duet/mnist_lightning/MNIST_Lightning_Syft_Data_Scientist.ipynb

Only works for CPU right now.

They are currently working on Plan which make remote training way faster.

We will resume our collaboration in 1-2 month time.

Best,
T.C

**Comment by Geeks-Sid** [comment]
Any updates on this @tchaton ?

**Reply by tchaton** [reply]
Hey @Geeks-Sid,

There is no current plan to push this further. If you think there is a strong desire from the community, please open a feature request on PySyft.

Best,
T.C
